### ISSUE TITLE
Rename the project

### ISSUE THREAD

**Issue by pierreozoux** [hackmdio/codimd#720]
As discussed is #579, this project is a fork, and the people that created this project, and the company that is trying to sell hosting of a closed source fork of this project would like to keep the name HackMD (which I totally understand, it is a great name :) )

But still, using the same name is confusing for new people coming to this project.

I propose we change the name to make it clear, and I have the sense that there is consensus about that in #579 .

As you know, "There are 2 hard problems in computer science: cache invalidation, *naming things*, and off-by-1 errors." ([Leon Bambrick](https://twitter.com/secretGeek/status/7269997868))

So first, what is HackMD?

In my opinion, it is the greatest pad with md support. Another amazing feature is the possibility to make presentations with revealjs!
So somehow, I guess the name should carry the following:
 - collaboration
 - text edition
 - md
 - presentation
 - pad (in reference to etherpad)
This is the first assumption to choose the name, so if you feel there are other values you want to convey, share here!

Then, as a second round, we should try to give names, and as a third round, let's vote about names!

Here is one idea about the name:
`ukuMD`
`ukubambisana` is `collaboration` in zulu, we would see the reference to hackMD, and it sounds great :)

Just an idea, waiting to hear yours!

**Comment by almereyda** [comment]
Most people seem to call Etherpads and HackMD pads either Pads or Hackpads, and I feel it could be a good choice to stick to it. Yet what makes HackMD outstanding, an almost a proper real-time wiki, is its Markdown support.

- `ukuMD` sounds a littly cryptic and is almost unpronouncable.

To go on with the brainstorm, allow me to throw those into the round:

- `Markpad`
- `Hackpad` sic! People will call it like that either way.
- `Notepad` in reference to ![notepad_1_16](https://user-images.githubusercontent.com/1645308/35970789-ee8a6118-0ccb-11e8-8bcb-b83ae5db85b8.png)

Let's try to stick with a pronouncable name that keeps in the minds of the readers. Yet effect described in the first sentence of this comment will most likely persist to remain.

**Comment by ccoenen** [comment]
I don't follow your argument by which either one would *have* to be renamed.

I'm not opposed to this discussion, but I don't see much benefit just from a rename. Feel free to post suggestions, but they'd have to be darn good to be worth the effort.

**Comment by pierreozoux** [comment]
You agreed to say it is not an open core model, yet the names are CE and EE which is usually used for open core model.
This is misleading, and you'll see, a lot of users (people that care about using only FLOSS, like me) will be disappointed, realizing that it is not an open core model (not saying that is my favorite, but better than a fork).

I personally don't care, as I already know what to expect.

And to say it one more time, I totally respect your desire to make a living from your project, and I respect your decision in finding the best way to do so.

You just have to understand that there are users that seek to only use FLOSS.
And when we pick a software, we prefer to pick one that will be maintained for the coming years.
(And yes, I do prefer to pick FLOSS that is backed by a company that is making good money, it means the software will be maintained.)
And the signal you send by making a fork is making us think that this will not be maintained.

We are part of the community, we developed this [package](https://github.com/indiehosters/hackmd/) for [libre.sh](http://libre.sh), our infrastructure. And that's why we care. We invested time in this (nothing compared to you, obviously). And maybe we'll have to start from scratch again trying to find a better alternative.

That's fine, it is life, not trying to put pressure on you and making you responsible for this. You made this beautiful open source project, and I take it as a gift, I don't expect anything from you besides that.
This is part of the game, and I accept the rules.

Back to the issue in itself, as @SISheogorath said, maybe HackMD/HackMDLibre is better.
I would take the responsibility to make the rename PR if you want.
I would even offer hosting for the libre demo version if you want, instead of heroku, then we would be a bit more part of the community :) ?

Or, let me try :) the best way to close this issue would be to choose an open core business model? :)

I know it is low priority and it feels bike shedding, but in my eyes, this is important too.
I let you close the issue if you feel it is irrelevant, I'm fine, you do as you want.

**Comment by ccoenen** [comment]
In case this was directed at me: I am doing a little moderation of issues or perhaps "community management" as some call it today. I contribute an occasional Pull Request but wasn't involved in the original development and I'm not involved in the EE in any way. The same goes for @SISheogorath as far as I know. In short: we don't have any say over the business model.

**Comment by SISheogorath** [comment]
Okay guys, let's keep calm and focused.

I think that the details of why everything is like it is right now are a bit too complex to point out here (before writing this answer, I tried and failed). The history why it is like it is right now is out of scope of this issue.

Let's check that's the purpose of this issue.
**Problem:**
There is Hackmd.io, HackMD EE and HackMD CE. The EE and CE confuses people, as it leads them to the false conclusion that this project is running the open core model.

**Solution:**
Removing CE and use another term to make it clear that CE is not .io and EE. Also it's not running the open core model.

**What is out of scope?**
All details why the model is like it is.

**What is the purpose of this issue?**
Finding a term or name that can be used instead of "CE" or "HackMD CE" so users no longer gets confused.

---

For all who are curious about why it is like it is, the community announcement explains very detailed, why this split was made. More details can be pointed out in a more interactive conversation like the Gitter or Matrix.org chat. I don't think there is really a point in writing all this down here as it's in various places, too, complex.

I hope, this helps to stay on our way to improve HackMD and find a solution that fits all interests :)

**Comment by almereyda** [comment]
So we are forking off and underline this step with a new name.

Sounds good to me. Thanks to the originators for their code donation to the
public, us.

I'd still be happy about Notepad, then I also wouldn't feel too bad about
building a real-time wiki from that.

Assuming here we self-invoke as a group of maintainers of the originally
donated code and herefore the question of this issue may rather be if we
have any aims where to bring this child and how it is delivered.

**Comment by Marwe** [comment]
`mdpad` or `madpad` come to my mind ;)

**Comment by RoyiAvital** [comment]
What about `CollabPad`?

Any chance offering this in the model of StackEdit?
With ability to Sync against Google Drive, OneDrive and DropBox?

Even better to have also Desktop version with ability to work against local `md` files.

Thank You.

**Comment by ccoenen** [comment]
> Any chance offering this in the model of StackEdit? [...]

Please create a new issue for suggestions. Nobody will find it in here or get back to you about this. If you create a fresh issue for it, it has a much higher chance of being picked up by somebody!

**Comment by almereyda** [comment]
Mdpad and Madpad sound nice, yet not mass compatible, CollabPad a little dowdy to me.

From Node-world we know some versioning practice that develops the next major release on branches called `next`, and the ownCloud community recently forked into Nextcloud, why I propose ...

- Nextpad

which is simple, easy to remember and kind of suits the case.

Unfortunately this title is already taken by [some Windows 10 program](http://www.definestudio.in/2016/04/nextpad.html). 

Did anyone propose

- Librepad

yet?

This term is only occupied by [programming exercises](https://github.com/xrisk/librepad) and [personal pet projects](https://github.com/coreseekdev/librepad).

**Comment by ccoenen** [comment]
Personally, I find "libre" as part of a name very cumbersome. This may be because in germany nobody knows how to pronounce it.

**Comment by ccoenen** [comment]
which does not explain how to pronounce it.

**Comment by SISheogorath** [comment]
> which does not explain how to pronounce it.

https://en.wiktionary.org/wiki/libre#Pronunciation

:wink:

**Comment by mathias-b** [comment]
I'm not fond of *Nextstuff-thing-bidule-truc*, sounds too hyped to me.
**Librepad** looks and sound great, specially in a fork context.

My 2c. :pray:

**Comment by m-blc** [comment]
Bumping this issue up. @pierreozoux are you still willing to submit a PR?

**Comment by ccoenen** [comment]
As long as we haven't settled on a name, a PR would be a tiny bit premature, don't you think?

**Comment by cben** [comment]
"Markpad" is taken:
- http://code52.org/DownmarkerWPF/ — too close, would be confusing
- http://brunofruchard.com/markpad.html

**Comment by almereyda** [comment]
How can we issue a vote?

In order of appearance in this issue

- HackMD
- ukuMD
- Hackpad
- Markpad
- Notepad
- HackMD CE
- HackMDLibre
- MDPad
- Madpad
- Nextpad
- Librepad

have been proposed before. Does anybody wish to withdraw their proposal?

To find a GitHub compliant voting mechanism, we could offer an official proposal timeframe, where we drop one proposal per comment into this thread, and then use the emoji reactions to generate quantitative responses. The highest number wins.

Else we just use the last proposal which did not produce any further counter proposals and close here.

---

As a side note, I am myself member of the cohort of German mother tongue speakers and also repeatingly managed to pronounce the word *libre* in the past. Therefore either I am nobody, or the statement above about the pronounceability of *libre* in Germany is wrong.

**Comment by ccoenen** [comment]
The last few weeks with many misdirected issues (those that were related to the hackmd.io outage, for example) convinced me that renaming is probably a good idea. I still think that we're on -very- different sides for whom or what we're giving this a new name.

In my opinion, slapping "next" or "libre" on top of something does not create a good name. Those are words for developers, devops or people with some kind of background in open source. These words mean nothing to the regular user. I am aiming for regular users.

A very similar gripe is with Markdown, MD, Mark, Down and Hack. Those are also not something I repeatedly want to bring up in a conversation with regular people. Yes, every single Web Developer knows what Markdown is. No, not a single "regular" person knows what this is. It adds nothing of value to the name. It only sounds weird.

Basically: If we're giving this a name, I would _love_ it to be something that _also_ works in these sentences:

- "You can look it up in (the) ______"
- "I sent you a link to the Party ______"
- "We're using (the) ____ to organize and the results will be mailed to each of you"
- "make a quick note to (the) ______"

Fill these sentences with "pad" or "wiki" and it mostly works for normal humans. Fill them with most of the suggestions we have right now, and you just sound weird.

I'd take names like "blender", "digikam", "darktable", "React", "Phaser", "OGRE", "WordPress" over any kind of "FreeSomething", "LibreSomething", "NextSomething".

**Comment by pierreozoux** [comment]
I kinda like `uku` and it fits the blanks :) It will stick in people's mind! and `uku.md` seems available ;)

**Comment by almereyda** [comment]
I've just asked a friend who considers himself "a regular person". He knows
what MarkDown is and considers it a superficial, simple thing to learn and
know about.

I also think a pad-like thingy should be called *pad.

Ukupad may be a little rough on the edges.

We could go all-in zen and just call it Pad.

**Comment by bumbleblue** [comment]
This is adding to the re-naming issue.
From a discussion over at prism-break: https://github.com/nylira/prism-break/issues/1829
Hope you guys can resolve this soon so more people can use hackmd (or whatever it will be called!).

**Comment by almereyda** [comment]
From the discussion at prism-break I am even more convinced we should move and rename the project into a libre community.

https://github.com/hackmdio/hackmd/blob/master/LICENSE clearly allows us to do so. The good thing is, we don't even need any permission to do so, but can just take the code as is, and create a libre fork of what is supposed to be the CF of HackMD.

I agree with the notion that the Libre* and Next* prefixes are only quite useful to technical communities. Yet I am still in favour of a more pronounceable and public compatible name.

`Pad` seemed to have received positive reactions above.

**Comment by SISheogorath** [comment]
First of all, we don't need to fork anything as this is already the free (libre) community version.

The only thing that is discussed here, if and how we should rename it. I don't see the point in doing a complete rename (like recently PIWIK did). Why? Because to me our existing users are more important than new groups that may want to adapt HackMD for whatever. We have no need of more feature requests,we need more people who implement stuff and they should be capable to find and help here if they want.

The CF vs. CE is that CE leads to the thought of an OpenCore model, which is not the case :)

**Comment by zeigerpuppy** [comment]
If someone using Markdown is a "marker", why not **MarkerPad**?

**Comment by ccoenen** [comment]
**Textmarker**?

**Comment by almereyda** [comment]
Textpad?

**Comment by pferreir** [comment]
HouseMD? :trollface:

**Comment by SISheogorath** [comment]
As I just thought a bit about it:

CodiMD, based on the latin word codicellus (or codicillus) which means notepad (or notebpads), is maybe a good new name. Those who really read the description will get it and those who don't may think we tried to cutify "Code" or "Coder" which is in one or another way close to Hack or Hacker, which than again is close to HackMD ^^

It also fits into the trend to use wise looking latin words in free software projects wiithout overused words like Libre.

Anyone against it?

**Comment by pferreir** [comment]
The only problem I can see is confusion with "Kodi", the media centre software. Otherwise, it sounds quite nice.

**Comment by ccoenen** [comment]
I like the direction of codi(cellus/cillus). It seems to be the diminuitive form of codex / **caudex**, so why not go with the "big" one? This also means [book, writing](https://en.m.wiktionary.org/wiki/caudex) but also [caudex](https://en.wikipedia.org/wiki/Caudex): "A caudex (plural: caudices) of a plant is a stem, but the term is also used to mean a rootstock and particularly a basal stem structure from which new growth arises."

putting it into the conversation test proposed above:

* "You can look it up in (the) caudex"
* "I sent you a link to the Party caudex"
* "We're using (the) caudex to organize and the results will be mailed to each of you"
* "make a quick note to (the) caudex"
* "just choose any appointment from (the) caudex"
* "add your name to (the) caudex over here [url]"

As far as I can tell, [no other project uses this word](https://alternativeto.net/browse/search?q=caudex).

**Comment by unteem** [comment]
I'm also in with the codi direction, CodiMD sounds quite good and makes sense.

I also wanted to add that the renaming is kind of starting to be critical. I have quite a number of users that are starting to become confused about functionalities that hackmd.io offers and what they can get on the self-hosted version. 

It feels like there is a community growing here, lets fork it ;)

**Comment by SISheogorath** [comment]
I'm going to rename the project to CodiMD at the end of the week. Along with that, we either make a decision about switching away from GitHub or not (See https://github.com/hackmdio/hackmd/issues/838).

At least that's my plan if there are no (significant) voices against it.

**Comment by SISheogorath** [comment]
I started the renaming in #850

**Comment by edoardoo** [comment]
I'm still trying to figure out how CodiMD won over Hackpad.  

I can't see my self filling any of the sentence proposed with CodiMD because: 
- **CodiMD pronunciation:** `[codi ɛm diː]` it's cumbersome and the two MarkDown letters interrupt the word (and the sentence) flow, which leads to shortenings
- **Shortening CodiMD**: Codi, which is very similar to Kodi the XBMC successor, which leads to confusion, beside the fact that, and that's just my taste, Codi doesn't sound cool. 
- **The CodiMD word root**: looks like a truncation of Coding, which is probably why the first time I've seen the repo name I've thought "Hey cool! A HackMD coding fork!". Obviously this was before finding this issue.  Also, even if I dind't like it and I wasn't that good at it, I actually studied latin in high school for 5 years. 

About Hackpad:
- sounds incredibly cool 😎
 "Wait a second, I wrote it on my hackpad" Hemingway
 "Let's have a hackpad later and fix the speech" Successful startupper
 "What? You don't know hackpad? It's like drive, but better" the cool kids at school
- There are some options for the domain that don't cost an eye (174€ was the cheapest after a quick search for .md) 

Sorry for being stupidly late, I totally lost the track on this project, I just loved from the beginning (as probably you all) and would've loved to see it gain more traction because it's open, because of the features and values and, why not, because it just sounds cool.
